crates/net/network/src/transactions.rs

```diff
@@ -472,10 +475,27 @@ where
         }
     }
 
-    fn report_bad_message(&self, peer_id: PeerId) {
-        trace!(target: "net::tx", ?peer_id, "Penalizing peer for bad transaction");
+    fn report_bad_message(&self, peer_id: PeerId, req_err: Option<RequestError>) {
```

Comment:
we can keep this and rename it to `on_request_error` and we need another function that accepts a `ReputationChangeKind`.
then this function here does not need an option anymore

Reply:
> Hey, just to make it clear, should I revert the changes on the report_bad_message

something in between, revert most changes for this but rename to `report_message(kind)` and add a `ReputationChangeKind` the new `on_request_error ` function then calls this

Reply:
I did some changes regarding your comment, did I get it right?